lib/internal/errors.js

```diff
@@ -1450,7 +1455,11 @@ E('ERR_MISSING_ARGS',
     return `${msg} must be specified`;
   }, TypeError);
 E('ERR_MISSING_OPTION', '%s is required', TypeError);
-E('ERR_MODULE_NOT_FOUND', (path, base, type = 'package') => {
+E('ERR_MODULE_NOT_FOUND', function(path, base, type = 'package') {
+  if (type === 'module') {
+    const { pathToFileURL } = lazyUrl();
+    this.url = pathToFileURL(path).href;
+  }
```

Comment:
Can you make it take a URL so we can ditch the circular dependency on `internal/url`?
```suggestion
E('ERR_MODULE_NOT_FOUND', function(path, base, type = 'package', url) {
  if (type === 'module') {
    setOwnProperty(this, 'url', url);
  }
```

Reply:
Sure, I've added that. It changes the algorithm slightly but seems good to me. I'm not sure what justification there is to use `setOwnProperty` though.